### ISSUE TITLE
AWS CodeBuild namespace_type change is ignored

### ISSUE THREAD

**Issue by byF** [terraform-providers/terraform-provider-aws#6214]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.8
+ provider.aws v1.40.0
```

### Affected Resource(s)

* aws_codebuild_project

### Terraform Configuration Files
This is the relevant part of the config, it is used inside a module:

Before change:
```hcl
variable "deployment_bucket" {}
variable "type" {}

resource "aws_codebuild_project" "test" {
  artifacts {
    type = "S3"
    location = "${var.deployment_bucket}"
    path     = "artifacts/DealWriter/${var.type}/"
  }
  ...
}
```

After adding `namespace_type`:
```hcl
variable "deployment_bucket" {}
variable "type" {}

resource "aws_codebuild_project" "test" {
  artifacts {
    type = "S3"
    location = "${var.deployment_bucket}"
    path     = "artifacts/DealWriter/${var.type}/"
   namespace_type = "BUILD_ID"
  }
  ...
}
```

### Expected Behavior

Terraform should determine there is a change in state and act accordingly.

### Actual Behavior

Terraform says there is no change and infrastructure is up-to-date, even though `terraform.tfstate` contains `"artifacts.537882814.namespace_type": "NONE",`

### Steps to Reproduce

1. `terraform get`
2. `terraform plan`

### Important Factoids

I've got this config in a module

### References

Might be not connected,  but I find it suspicious there are more than one thing broken in `artifacts` definition:

* #6127

**Comment by erikpaasonen** [comment]
Also appears to be related: https://github.com/terraform-providers/terraform-provider-aws/issues/2050

**Comment by bflad** [comment]
The fixes for having the `aws_codebuild_project` resource `artifacts` configuration block perform better drift detection and updates have been merged and will release with version 2.22.0 of the Terraform AWS Provider, tomorrow. 👍
